### DISCUSSION TITLE
Directories missing from .tar.gz file

### DISCUSSION THREAD

**Discussion by deepali0162** [python-poetry/poetry#6359] [Q&A]
I am following https://github.com/python-poetry/poetry to create my pyproject.toml file.
My folder structure looks like below:

>example
>>pyproject.toml
>>src
>>>example_package
>>>>config
>>>>logs
>>>>__init__.py
>>>>app.py

My pyproject.toml looks like below:
```
[build-system]
requires = ["setuptools>=61.0"]
build-backend = "setuptools.build_meta"

[tool.poetry]
name = "example_package"
version = "0.1.1"
requires_python = ">=3.7"

packages = [
    {include = "markets_package", from = "src"}
]

classifiers = [
     "Programming Language :: Python :: 3",
    "License :: OSI Approved :: MIT License",
    "Operating System :: OS Independent",
]

dependencies = [
"numpy==1.21.6.*"
]
```

When I execute below command to create package file, the .tar.gz is missing config and logs folder. Please advise how this could be sorted.
`python -m build`

**Comment by neersighted** [comment]
You need to explicitly include them in the build of your distribution files:

https://python-poetry.org/docs/pyproject/#include-and-exclude

Additionally, `requires_python` should not be used -- instead use `tool.poetry.dependencies.python` to set that value. Poetry will infer the correct `Programming Language :: Python` classifier values from that declaration.

**Comment by deepali0162** [comment]
Thanks for replying but could you please clarify?
Does it mean it should be :
```
[tool.poetry]
.....
packages = [
    {include = "markets_package/config/config.json", from = "src"},
    {include = "markets_package/config/config2.json", from = "src"}
]
```
or
```
[tool.poetry]
......
include = [
    { path = "markets_package/config/config.json" },
    { path = "markets_package/config/config2.json" }
]
```
Well, I tried both ways and my config folder is still missing.

**Reply by neersighted** [reply]
Please refer to the documentation -- `packages =` is for Python modules, and `include =` is for non-module files. `include = ["path/to/file/from/repo/root"]` is the syntax that should work for you here.

**Comment by deepali0162** [comment]
I have updated my code to:
```
include = ["example_package/config/config.json"]
include = ["example_package/config/config2.json"]
```

I have also tried with complete path however, it is still not working. Is it an issue with project structure?

**Reply by neersighted** [reply]
You can only have one `include =` key in your project file -- I am not sure what duplicates will do, but it is a violation of the TOML spec. Instead make each path a entry in the list.

Also, configuration really should not live in your python package. That's unconventional/not going to work with most Python packaging mechanisms. That being said, any files in `markets_package` really should be getting picked up. I'm not sure I can do much to help you without a sanitized repo or similar to clone locally and test.
